controllers/ocmagent/ocmagent_controller.go

```diff
@@ -0,0 +1,199 @@
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.2/pkg/reconcile
+func (r *OcmAgentReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling OCMAgent")
+
+	// Fetch the OCMAgent instance
+	instance := ocmagentv1alpha1.OcmAgent{}
+	err := r.Client.Get(ctx, request.NamespacedName, &instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			localmetrics.UpdateMetricOcmAgentResourceAbsent()
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		log.Error(err, "Failed to retrieve OCMAgent. Will retry on next reconcile.")
+		return reconcile.Result{}, err
+	}
+	localmetrics.ResetMetricOcmAgentResourceAbsent()
+	oaohandler, err := r.OCMAgentHandlerBuilder.New()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Is the OCMAgent being deleted?
+	if !instance.DeletionTimestamp.IsZero() {
+		log.V(2).Info("Entering EnsureOCMAgentResourcesAbsent")
+		err := oaohandler.EnsureOCMAgentResourcesAbsent(instance)
+		if err != nil {
+			log.Error(err, "Failed to remove OCMAgent. Will retry on next reconcile.")
+			return reconcile.Result{}, err
+		}
+		// The finalizer can now be removed
+		if controllerutil.ContainsFinalizer(&instance, ctrlconst.ReconcileOCMAgentFinalizer) {
+			controllerutil.RemoveFinalizer(&instance, ctrlconst.ReconcileOCMAgentFinalizer)
+			if err := r.Client.Update(ctx, &instance); err != nil {
+				log.Error(err, "Failed to remove finalizer from OCMAgent resource. Will retry on next reconcile.")
+				return reconcile.Result{}, err
+			}
+		}
+		log.Info("Successfully removed OCMAgent resources.")
+	} else {
+		// There needs to be an OCM Agent
+		log.V(2).Info("Entering EnsureOCMAgentResourcesExist")
+		err := oaohandler.EnsureOCMAgentResourcesExist(instance)
+		if err != nil {
+			log.Error(err, "Failed to create OCMAgent. Will retry on next reconcile.")
+			return reconcile.Result{}, err
+		}
+
+		// The OCM Agent is deployed, so set a finalizer on the resource
+		if !controllerutil.ContainsFinalizer(&instance, ctrlconst.ReconcileOCMAgentFinalizer) {
+			controllerutil.AddFinalizer(&instance, ctrlconst.ReconcileOCMAgentFinalizer)
+			if err := r.Client.Update(ctx, &instance); err != nil {
+				log.Error(err, "Failed to apply finalizer to OCMAgent resource. Will retry on next reconcile.")
+				return reconcile.Result{}, err
+			}
+		}
+		log.Info("Successfully setup OCMAgent resources.")
+	}
+
+	return reconcile.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *OcmAgentReconciler) SetupWithManager(mgr ctrl.Manager) error {
```

Comment:
I _think_ this can be simplified to just:

```
	return ctrl.NewControllerManagedBy(mgr).
		For(&ocmagentv1alpha1.OcmAgent{}).
		Owns(&netv1.NetworkPolicy{}).
		Owns(&appsv1.Deployment{}).
		Owns(&corev1.Service{}).
		Owns(&corev1.Secret{}).
		Owns(&corev1.ConfigMap{}).
		Complete(r)
```